image.h

```diff
@@ -17,21 +17,37 @@ class Image {
 	bool imageSizeIsValid() {
 		return m_rows <= 1024 && m_columns <= 1024;
 	}
-	uint16_t pixelIndex(uint16_t x, uint16_t y) {
+	uint16_t pixelIndex(const uint16_t x, const uint16_t y) {
 		return x * m_columns + y;
 	}
-	// Change the name of this function... Try a more specific name instead of 'get'
-	uint8_t getPixel(uint16_t x, uint16_t y) {
+	uint8_t getPixelAtIndex(const uint16_t x, const uint16_t y) {
 		return m_pixels[pixelIndex(x, y)];
 	}
-	void scanPixels(std::function<void(uint8_t, uint16_t, uint16_t)> takePixel) {
+	void scanPixels(std::function<void(const uint8_t, const uint16_t, const uint16_t, const uint16_t)> takePixel) {
 		for (int x = 0; x < m_rows; x++) {
 			for (int y = 0; y < m_columns; y++) {
-				takePixel(getPixel(x, y), x, y);
+				takePixel(getPixelAtIndex(x, y), x, y, pixelIndex(x,y));
 			}
 		}
 	}
 	const uint16_t m_rows;
 	const uint16_t m_columns;
 	uint8_t* m_pixels; // max 1k x 1k image
 };
+
+class RawImage {
+public:
+	RawImage(std::shared_ptr<Image> rawImage) {	
+		m_internalImage = rawImage;
+	}
+	std::shared_ptr<Image> m_internalImage;
+};
+
+class BrightenedImage {
+public:
+	BrightenedImage(std::shared_ptr<Image> brightenedImage) {
+		m_internalImage = brightenedImage;
+	}
+	std::shared_ptr<Image> m_internalImage;
+	int m_attenuatedPixelCount = 0;
```

Comment:
Public member makes this mutable and non-scoped. It only needs to be set at initialisation time, right? After that never should change. Is it hard to impose that restriction via type-safety?